src/Persistence_matrix/include/gudhi/persistence_interval.h

```diff
@@ -0,0 +1,215 @@
+/*    This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+ *    See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+ *    Author(s):       Hannah Schreiber
+ *
+ *    Copyright (C) 2024 Inria
+ *
+ *    Modification(s):
+ *      - YYYY/MM Author: Description of the modification
+ */
+
+/**
+ * @file persistence_interval.h
+ * @author Hannah Schreiber
+ * @brief Contains @ref Gudhi::persistence_matrix::Persistence_interval class.
+ */
+
+#ifndef PM_INTERVAL_INCLUDED
+#define PM_INTERVAL_INCLUDED
+
+#include <ostream>  //std::ostream
+#include <limits>   //std::numeric_limits
+#include <tuple>
+#include <utility>  //std::move
+
+namespace Gudhi {
+namespace persistence_matrix {
+
+/**
+ * @class Persistence_interval persistence_interval.h gudhi/persistence_interval.h
+ * @ingroup persistence_matrix
+ *
+ * @brief Type for an interval in a persistent diagram or barcode. Stores the birth, death and dimension of the
+ * interval. It can be used as a tuple with `get`/`std::get` (birth, death and dimension in this order),
+ * `std::tuple_element` and `std::tuple_size`, as well as structured binding.
+ * 
+ * @tparam dimension_type Type of the dimension value.
+ * @tparam event_value_type Type of the birth and death value.
+ */
+template <typename dimension_type, typename event_value_type>
+struct Persistence_interval {
+  /**
+   * @brief Stores the infinity value for birth and death events. Its value depends on the template parameter
+   * `event_value_type`:
+   * - if `event_value_type` has a native infinity value, it takes this value,
+   * - otherwise, if `event_value_type` is a signed type, it takes value -1,
+   * - otherwise, if `event_value_type` is a unsigned type, it takes the maximal possible value.
+   *
+   * Is also used as default value for birth and death attributes when not initialized.
+   */
+  static constexpr event_value_type inf = std::numeric_limits<event_value_type>::has_infinity
+                                              ? std::numeric_limits<event_value_type>::infinity()
+                                              : static_cast<event_value_type>(-1);
+
+  /**
+   * @brief Constructor.
+   * 
+   * @param dim Dimension of the cycle. Default value: -1.
+   * @param birth Birth value of the cycle. Default value: @ref inf.
+   * @param death Death value of the cycle. Default value: @ref inf.
+   */
+  Persistence_interval(dimension_type dim = -1, event_value_type birth = inf, event_value_type death = inf)
+      : dim(dim), birth(birth), death(death) {}
+
+  dimension_type dim;     /**< Dimension of the cycle.*/
+  event_value_type birth; /**< Birth value of the cycle. */
+  event_value_type death; /**< Death value of the cycle. */
+
+  inline friend std::ostream &operator<<(std::ostream &stream, const Persistence_interval &interval) {
+    stream << "[" << interval.dim << "] ";
+    stream << interval.birth << ", " << interval.death;
```

Comment:
`Inf` values should be detected and print something else.
```cpp
  using Dimension_type = typename Matrix::dimension_type;
  using Pos_index = typename Matrix::pos_index;
  using Interval = Gudhi::persistence_matrix::Persistence_interval<Dimension_type, Pos_index>;

  Interval bar_default;
  std::cout << bar_default << std::endl;
  // prints: [-1] 4294967295, 4294967295
```

Reply:
Maybe the same remark than Marc, but:
```cpp
  Interval bar(0,1,2);
  std::cout << bar << std::endl;
  // print: [0] 1, 2
  std::cout << std::get<0>(bar) << ", " << std::get<1>(bar) << ", " << std::get<2>(bar) << std::endl;
  // print: 1, 2, 0
```

Reply:
I am not sure to understand which remark of Marc you are referring to?

Reply:
Well, I already explained why it was like that, I was waiting for an answer. So, should I change the order in the constructor to match the `std::get`s? But for the print, I like it better to have the dimension first. `[0] 1, 2` is much easier to read than `1, 2 [0]` in particular when there are a lot of bars.